elyra/pipeline/component_registry.py

```diff
@@ -255,14 +267,17 @@ def get_component(self, component_id: str) -> Optional[Component]:
                                  if component.id == component_id), None)
         return cached_component
 
-    def get_categories(self) -> List[Dict]:
+    def get_all_categories(self) -> List[ComponentCategory]:
         if self._is_cache_expired():
             self._update_cache()
 
         return self._cached_categories
 
     def _update_cache(self):
-        self._cached_components, self._cached_categories = super().get_all_components()
+        registry = super()._read_component_registry()
+        self._cached_components = registry.get('components')
+        self._cached_categories = registry.get('categories')
```

Comment:
I'm not able to place a comment on `CachedComponentRegistry.get_component()` directly, so adding one here.

Per my previous comment, I'm fairly certain that `CachedComponentRegistry.get_component()` is returning a different result than `ComponentRegistry.get_component()` because this one is behaving as I'd suspect and just using the results of `get_all_components()` to locate the specific component.

Reply:
Per Kiersten's previous response, my assumption was incorrect.  This is moot.